src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp

```diff
@@ -145,6 +146,7 @@ bool ngraph::pass::MOCTransformations::run_on_model(const std::shared_ptr<ngraph
     common_fusions->add_matcher<ngraph::pass::HSigmoidFusion>();
     common_fusions->add_matcher<ngraph::pass::NormalizeL2Fusion>();
     common_fusions->add_matcher<ngraph::pass::ClampFusion>();
+    common_fusions->add_matcher<ov::pass::OptimizerGatherND>();
```

Comment:
This transformation is more like decomposition than fusion, and static shapes are needed.
My suggestion would be to add the matcher to the CommonDecompositions group.
https://github.com/openvinotoolkit/openvino/blob/b98e5c7256015518f8ff71a0a590e2ef5478c1dd/src/common/transformations/src/transformations/common_optimizations/common_optimizations.cpp#L140-L141